src/app/core/components/header/header.component.ts

```diff
@@ -1,10 +1,38 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, OnInit, Input } from "@angular/core";
+import { TimeTableService } from "../../../timetable/services/time-table.service";
 
 @Component({
-  selector: 'app-header',
-  templateUrl: './header.component.html',
-  styleUrls: ['./header.component.scss']
+  selector: "app-header",
+  templateUrl: "./header.component.html",
+  styleUrls: ["./header.component.scss"]
 })
 export class HeaderComponent {
-  @Input() weekDay: string;
+  weekDay: string = "Monday";
+  constructor(private _timetableService$: TimeTableService) {}
+  ngOnInit() {
+    this._timetableService$.message.subscribe(msg => {
```

Comment:
1. create a new property in this class like `public subscriptions$ = new Subscription();`
2. then include OnDestroy and in ngOnDestroy put `this.subscriptions$.unsubscribe();`
3. at this line put this subscription in `subscription$` using `add` method